source/common/protobuf/utility.cc

```diff
@@ -12,10 +12,17 @@ MissingFieldException::MissingFieldException(const std::string& field_name,
           fmt::format("Field '{}' is missing in: {}", field_name, message.DebugString())) {}
 
 void MessageUtil::loadFromFile(const std::string& path, Protobuf::Message& message) {
-  const std::string json = Filesystem::fileReadToEnd(path);
-  const auto status = Protobuf::util::JsonStringToMessage(json, &message);
+  const std::string file_contents = Filesystem::fileReadToEnd(path);
+  // If the filename ends with .pb, do a best-effort attempt to parse it as a proto.
+  if (path.size() > 3 && path.substr(path.size() - 3) == ".pb") {
+    bool status = message.ParseFromString(file_contents);
+    if (status) {
```

Comment:
Failure case falls through "by design" because if we don't attempt to parse *.pb as json we fail some existing tests.  Which IMO is terrible.   I can go and fix our tests but I can't fix any equivalent silliness if other people's current configs have json in *.pb files, so I opted to leave the fallthrough in.

If it's OK to make this a hard failure let me know and I'll do that and go fix the tests :-)